contracts/EVMScriptFactories/ActivateNodeOperators.sol

```diff
@@ -0,0 +1,155 @@
+    string private constant MANAGER_ALREADY_HAS_ROLE = "MANAGER_ALREADY_HAS_ROLE";
+    string private constant NODE_OPERATORS_IS_NOT_SORTED = "NODE_OPERATORS_IS_NOT_SORTED";
+
+    // -------------
+    // CONSTRUCTOR
+    // -------------
+
+    constructor(
+        address _trustedCaller,
+        address _nodeOperatorsRegistry,
+        address _acl
+    ) TrustedCaller(_trustedCaller) {
+        nodeOperatorsRegistry = INodeOperatorsRegistry(_nodeOperatorsRegistry);
+        acl = IACL(_acl);
+    }
+
+    // -------------
+    // EXTERNAL METHODS
+    // -------------
+
+    function createEVMScript(
+        address _creator,
+        bytes memory _evmScriptCallData
+    ) external view override onlyTrustedCaller(_creator) returns (bytes memory) {
+        ActivateNodeOperatorInput[] memory decodedCallData = _decodeEVMScriptCallData(
+            _evmScriptCallData
+        );
+
+        _validateInputData(decodedCallData);
+
+        address[] memory toAddresses = new address[](decodedCallData.length * 2);
+        bytes4[] memory methodIds = new bytes4[](decodedCallData.length * 2);
+        bytes[] memory encodedCalldata = new bytes[](decodedCallData.length * 2);
+
+        for (uint256 i = 0; i < decodedCallData.length; i++) {
+            toAddresses[i * 2] = address(nodeOperatorsRegistry);
+            methodIds[i * 2] = ACTIVATE_NODE_OPERATOR_SELECTOR;
+            encodedCalldata[i * 2] = abi.encode(decodedCallData[i].nodeOperatorId);
+
+            uint256[] memory permissionParams = new uint256[](1);
+            permissionParams[0] = (1 << 240) + decodedCallData[i].nodeOperatorId;
+
+            toAddresses[i * 2 + 1] = address(acl);
+            methodIds[i * 2 + 1] = GRANT_PERMISSION_P_SELECTOR;
+            encodedCalldata[i * 2 + 1] = abi.encode(
+                decodedCallData[i].managerAddress,
+                address(nodeOperatorsRegistry),
+                MANAGE_SIGNING_KEYS_ROLE,
+                permissionParams
+            );
+        }
+
+        return EVMScriptCreator.createEVMScript(toAddresses, methodIds, encodedCalldata);
+    }
+
+    function decodeEVMScriptCallData(
+        bytes memory _evmScriptCallData
+    ) external pure returns (ActivateNodeOperatorInput[] memory) {
+        return _decodeEVMScriptCallData(_evmScriptCallData);
+    }
+
+    // ------------------
+    // PRIVATE METHODS
+    // ------------------
+
+    function _decodeEVMScriptCallData(
+        bytes memory _evmScriptCallData
+    ) private pure returns (ActivateNodeOperatorInput[] memory) {
+        return abi.decode(_evmScriptCallData, (ActivateNodeOperatorInput[]));
+    }
+
+    function _validateInputData(
+        ActivateNodeOperatorInput[] memory _activateNodeOperatorInputs
+    ) private view {
+        uint256 nodeOperatorsCount = nodeOperatorsRegistry.getNodeOperatorsCount();
+        for (uint256 i = 0; i < _activateNodeOperatorInputs.length; i++) {
```

Comment:
It requires to run all motions on live network to get precise estimations, as far as brownie supports only berlin hardfork. Will add limits after that

Reply:
I don't think we need such limits on the contract level because the gas costs may change due to external factors: updates of the block gas limit and EVM operations gas costs or node operators registry implementation changes. I think it's better to limit it on the frontend side + provide the batch size recommendations in the guide for Simple DVT EasyTrack.